rust/lance/src/index/prefilter.rs

```diff
@@ -119,49 +122,59 @@ impl DatasetPreFilter {
         // This can only be computed as an allow list, since we have no idea
         // what the row ids were in the missing fragments.
 
-        // For each fragment, compute which row ids are still in use.
-        let dataset_ref = dataset.as_ref();
+        let path = dataset
+            .base
+            .child(format!("row_id_mask{}", dataset.manifest().version));
+
+        let session = dataset.session();
+
+        async fn load_row_ids_and_deletions(
+            dataset: &Dataset,
+        ) -> Result<Vec<(Arc<RowIdSequence>, Option<Arc<DeletionVector>>)>> {
+            stream::iter(dataset.get_fragments())
+                .map(|frag| async move {
+                    let row_ids = load_row_id_sequence(dataset, frag.metadata());
+                    let deletion_vector = frag.get_deletion_vector();
+                    let (row_ids, deletion_vector) = join!(row_ids, deletion_vector);
+                    Ok::<_, crate::Error>((row_ids?, deletion_vector?))
+                })
+                .buffer_unordered(10)
```

Comment:
I see you  just moved this function but maybe let's get rid of the random `10`?

Reply:
What is our best practice right now. I don't love this, but I just don't know what our current preferred pattern is right now.

Reply:
Since it's mostly I/O driven I'd recommend `dataset.object_store().io_parallelism()`?